clang_format.py

```diff
@@ -115,29 +161,63 @@ def clang_format_path() -> Path:
     return clang_format_file
 
 
-def main() -> None:
+def get_version_list(
+    versionmap: Mapping[
+        Tuple[int, int, int],
+        Mapping[str, str],
+    ]
+) -> Tuple[str, ...]:
+    data = [tuple(map(str, tup)) for tup in versionmap.keys()]
+    return tuple([".".join(version) for version in data])
+
+
+def main(argv: Optional[Sequence[str]] = None) -> Union[int, None]:  # type: ignore
```

Comment:
Removing the `# type: ignore` raises an important warning:

```
clang_format.py:174: error: Missing return statement  [return]
```

That is a completely valid warning: you need to have a return statement on each path through the function.

All you need is `return None` at the bottom.

Even better, change the return type to `int` and `return 0`.

_Even better_, propagate the return code from `clang-format` with `return subprocess.run(...).returncode`

Reply:
Thanks, @mattyclarkson  for the suggestion I have now updated the main function to return the status code of the run.